query/sources/impl/src/main/java/org/codice/ddf/admin/sources/opensearch/OpenSearchSourceUtils.java

```diff
@@ -103,33 +104,32 @@ public OpenSearchSourceUtils(ConfiguratorSuite configuratorSuite) {
             taskList.add(callables);
         }
 
-        PrioritizedBatchExecutor<ReportWithResultImpl<OpenSearchSourceConfigurationField>, ReportWithResultImpl<OpenSearchSourceConfigurationField>>
+        PrioritizedBatchExecutor<Report<OpenSearchSourceConfigurationField>, Report<OpenSearchSourceConfigurationField>>
                 prioritizedExecutor = new PrioritizedBatchExecutor(THREAD_POOL_SIZE,
                 taskList,
                 new SourceTaskHandler<OpenSearchSourceConfigurationField>());
 
-        Optional<ReportWithResultImpl<OpenSearchSourceConfigurationField>> result =
+        Optional<Report<OpenSearchSourceConfigurationField>> result =
                 prioritizedExecutor.getFirst();
 
         if (result.isPresent()) {
             return result.get();
         } else {
-            return new ReportWithResultImpl<OpenSearchSourceConfigurationField>().addArgumentMessage(
-                    unknownEndpointError(hostField.path()));
+            return Reports.from(unknownEndpointError(hostField.path()));
         }
     }
 
-    public ReportWithResultImpl<OpenSearchSourceConfigurationField> getOpenSearchConfigFromUrl(
+    public Report<OpenSearchSourceConfigurationField> getOpenSearchConfigFromUrl(
             UrlField urlField, CredentialsField creds) {
-        ReportWithResultImpl<ResponseField> responseResult = requestUtils.sendGetRequest(urlField,
+        Report<ResponseField> responseResult = requestUtils.sendGetRequest(urlField,
                 creds,
                 GET_CAPABILITIES_PARAMS);
 
-        if (responseResult.containsErrorMsgs()) {
-            return (ReportWithResultImpl) responseResult;
+        if (responseResult.containsErrorMessages()) {
+            return (Report) responseResult;
```

Comment:
Again, we can't do `Reports.from(Report)` but we can get around it by doing `Reports.from(List<ErrorMessages>)`